sdk/src/loom/ModestMaps/Map.ls

```diff
@@ -0,0 +1,713 @@
+         * <p>Animating/tweening grid.scale fires START_ZOOMING, and STOP_ZOOMING 
+         * MapEvents unless you call grid.prepareForZooming first. Be sure
+         * to also call grid.stopZooming at the end of your animation.
+         *
+         * @param dir the direction of zoom, generally 1 for zooming in, or -1 for zooming out
+         * 
+         */ 
+        protected function zoomBy(dir:int):void
+        {
+            if (!grid.panning) {
+                var target:Number = dir < 0 ? Math.floor(grid.zoomLevel+dir) : Math.ceil(grid.zoomLevel+dir);
+                grid.zoomLevel = Math.min(Math.max(grid.minZoom, target), grid.maxZoom);
+            }
+        } 
+        
+       /**
+        * Add a marker at the given location (lat, lon)
+        *
+        * @param    Location of marker.
+        * @param    optionally, a sprite (where sprite.name=id) that will always be in the right place
+        */
+        public function putMarker(location:Location, marker:DisplayObject=null):void
+        {
+            markerClip.attachMarker(marker, location);
+        }
+
+        /**
+         * Get a marker with the given id if one was created.
+         *
+         * @param    ID of marker, opaque string.
+         */
+        public function getMarker(id:String):DisplayObject
+        {
+            return markerClip.getMarker(id);
+        }
+
+       /**
+        * Remove a marker with the given id.
+        *
+        * @param    ID of marker, opaque string.
+        */
+        public function removeMarker(id:String):void
+        {
+            markerClip.removeMarker(id); // also calls grid.removeMarker
+        }
+        
+        public function removeAllMarkers():void {
+            markerClip.removeAllMarkers();
+        }
+        
+       /**
+        * Dispatches MapEvent.EXTENT_CHANGED when the map is recentered.
+        * The MapEvent includes the new extent.
+        * 
+        * TODO: dispatch this on resize?
+        * TODO: should we move Map to loom.modestmaps.core so that this could be made internal instead of public?
+        *
+        * @see loom.modestmaps.events.MapEvent.EXTENT_CHANGED
+        */
+        protected function onExtentChanged(event:Event=null):void
+        {
+            if (hasEventListener(MapEvent.EXTENT_CHANGED)) {
+                dispatchEvent(new MapEvent(MapEvent.EXTENT_CHANGED, [getExtent()]));
+            }
+        }
+
+       /**
+        * Dispatches MapEvent.BEGIN_EXTENT_CHANGE when the map is about to be resized.
+        * The MapEvent includes the current.
+        *
+        * @see loom.modestmaps.events.MapEvent.BEGIN_EXTENT_CHANGE
+        */
+        protected function onExtentChanging():void
+        {
+            if (hasEventListener(MapEvent.BEGIN_EXTENT_CHANGE)) {
+                dispatchEvent(new MapEvent(MapEvent.BEGIN_EXTENT_CHANGE, [getExtent()]));
+            }
+        }
+
+
+
+        //NOTE_TEC: Loom doesn't support MouseWheel input at this time
```

Comment:
I think it does get wheel events actually? https://github.com/LoomSDK/LoomSDK/tree/dfe41a7b508b23c4695a40023ebd523b1c4ea655/docs/examples/ScrollWheelExample